src/Microsoft.ML.Core/Data/RoleMappedSchema.cs

```diff
@@ -98,68 +108,109 @@ public sealed class RoleMappedSchema
         private const string NameString = "Name";
         private const string FeatureContributionsString = "FeatureContributions";
 
+        /// <summary>
+        /// Instances of this are the keys of a <see cref="RoleMappedSchema"/>. This class also some holds important
+        /// commonly used pre-defined instances available (e.g., <see cref="Label"/>, <see cref="Feature"/>) that should
+        /// be used when possible for consistency reasons. However, practitioners should not be afraid to declare custom
+        /// roles if approppriate for their task.
+        /// </summary>
         public struct ColumnRole
         {
+            /// <summary>
+            /// Role for features. Commonly used as the independent variables given to trainers, and scorers.
+            /// </summary>
             public static ColumnRole Feature => FeatureString;
+
+            /// <summary>
+            /// Role for labels. Commonly used as the dependent variables given to trainers, and evaluators.
+            /// </summary>
             public static ColumnRole Label => LabelString;
+
+            /// <summary>
+            /// Role for group ID. Commonly used in ranking applications, for defining query boundaries, or
+            /// sequence classification, for defining the boundaries of an utterance.
+            /// </summary>
             public static ColumnRole Group => GroupString;
+
+            /// <summary>
+            /// Role for sample weights. Commonly used to point to a number to make trainers give more weight
+            /// to a particular example.
+            /// </summary>
             public static ColumnRole Weight => WeightString;
+
+            /// <summary>
+            /// Role for sample names. Useful for informational and tracking purposes when scoring, but typically
+            /// without affecting results.
+            /// </summary>
             public static ColumnRole Name => NameString;
+
+            // REVIEW: Does this really belong here?
+            /// <summary>
+            /// Role for feature contributions. Useful for specific diagnostic functionality.
+            /// </summary>
             public static ColumnRole FeatureContributions => FeatureContributionsString;
```

Comment:
>FeatureContributions [](start = 37, length = 20)

I don't really feel like this actually belongs here... certainly in ML.NET there are no usages, and the usages elsewhere are heavily localized so putting a custom role in the basic class is sort of odd... it would be better to have the score-roles.